integrated_channels/canvas/views.py

```diff
@@ -0,0 +1,112 @@
+"""
+
+from django.conf import settings
+from django.apps import apps
+import requests
+from rest_framework.response import Response
+from rest_framework import generics, status
+from six.moves.urllib.parse import urljoin
+
+from enterprise.utils import get_enterprise_customer
+from integrated_channels.canvas.models import CanvasEnterpriseCustomerConfiguration
+class CanvasCompleteOAuthView(generics.ListAPIView):
+    """
+        **Use Cases**
+
+            Retrieve and save a Canvas OAuth refresh token after an enterprise customer
+            authorizes to integrated courses.
+
+        **Example Requests**
+
+            GET /canvas/oauth-complete?code=123abc&state=abc123
+
+        **Query Parameters for GET**
+
+            * code: The one time use string generated by the Canvas API used to fetch the
+            access and refresh tokens for integrating with Canvas.
+
+            * state: The user's enterprise customer uuid used to associate the incoming
+            code with an enterprise configuration model.
+
+        **Response Values**
+
+            If the request for information about the course list is successful, an HTTP 200 "OK" response
+            is returned.
+
+            If code is not provided, a 400 error is returned.
+
+            If state is not provided, a 400 error is returned.
+
+            If the specified state is not valid or contained in the set of registered enterprise customers
+            a 404 error is returned.
+
+    """
+    def get(self, request, *args, **kwargs):  # pylint: disable=arguments-differ
+        app_config = apps.get_model('canvas', 'CanvasGlobalConfiguration').current()
+        canvas_oauth_token_path = app_config.oauth_token_auth_path
+        client_code = request.GET.get('code')
+        enterprise_customer_uuid = request.GET.get('state')
+        if not enterprise_customer_uuid:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    "message": (
+                        u"Enterprise ID required to integrate with Canvas."
+                    )
+                })
+
+        if not client_code:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    "message": (
+                        u"Canvas code token is required to complete the Oauth process."
+                    )
+                })
+
+        enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+        if not enterprise_customer:
+            return Response(
+                status=status.HTTP_404_NOT_FOUND,
+                data={
+                    "message": (
+                        u"No enterprise data found for given ciid."
+                    )
+                })
+
+        enterprise_config = CanvasEnterpriseCustomerConfiguration.objects.get(
+            enterprise_customer=enterprise_customer
+        )
+
+        print(enterprise_config)
```

Comment:
I assume you will remove the print statements before merge? However, if these are useful to log, you can use the LOGGER for more controlled logging

Reply:
Yikes, this was definitely a left over from debugging. Will be removed ASAP. but also interesting point on logger- may want to add some for clarity in future debugging. Thoughts/opinions on saving logs throughout the process? Ie "initializing client", "received authorization code from user" and/or "retrieved user refresh token, client is now ready to be instantiated" or others to the same effect?

Reply:
My usual inclination is to log any 'non essential' items at debug level only to avoid noise. If a step seems really important though, we can have it logged at info maybe. Some things also merit a warn level logging. But failures should certainly be thrown (best) or at least logged (if you are logging) with pertinent information we can view in the logs. I have not used newrelic specifically for that (I got used to sentry logging) but I assume it will work fine if we log appropriately

Reply:
I have more experience with sentry logging as well. I'll look through as see if anything is worth logging, otherwise the print statement is gone-zo